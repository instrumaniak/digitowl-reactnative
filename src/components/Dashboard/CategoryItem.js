import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

import { colors } from '../../config'

const CategoryItem = ({
  iconText,
  title,
  description,
  value
}) => (
  <View style={styles.container}>
    <View style={styles.iconContainer}>
      <Text style={styles.iconText}>{iconText}</Text>
    </View>
    <View style={styles.dataContainer}>
      <View>
        <Text style={styles.title}>{ title }</Text>
        <Text style={styles.description}>{ description }</Text>
      </View>
      <View>
        <Text style={styles.value}>{ value }</Text>
      </View>
    </View>
  </View>
)

CategoryItem.propTypes = {
  iconText: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  value: PropTypes.string
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  iconContainer: {
    width: 50,
    height: 50,
    backgroundColor: colors.grey,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center'
  },
  iconText: {
    color: colors.white,
    //fontWeight: 'bold',
    fontSize: 35
  },
  dataContainer: {
    flex: 1,
    paddingLeft: 15,
    flexDirection: 'row',
    justifyContent: 'space-between'
  },

  // typography
  title: {
    color: colors.grey1,
    fontSize: 20,
    //fontWeight: 'bold'
  },
  description: {
    color: colors.grey,
    fontSize: 16
  },
  value: {
    color: colors.grey1,
    fontSize: 18,
    fontWeight: 'bold'
  }
})

export default CategoryItem
