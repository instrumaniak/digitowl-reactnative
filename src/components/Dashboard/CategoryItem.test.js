/**
 *   Tests for component: CategoryItem
 */

import React from 'react'
import CategoryItem from './CategoryItem'
import renderer from 'react-test-renderer'

describe('CategoryItem component', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<CategoryItem />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
