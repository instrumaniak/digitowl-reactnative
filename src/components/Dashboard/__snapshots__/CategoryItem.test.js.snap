// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem component Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c3c6c7",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0.355469,
          "lineHeight": 34,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 15,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#5B6267",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "400",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          }
        }
      />
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#c3c6c7",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "letterSpacing": -0.24,
            "lineHeight": 20,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#4B5358",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "600",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          }
        }
      />
    </View>
  </View>
</View>
`;
